lib/msf/core/opt_enum.rb

```diff
@@ -44,6 +53,10 @@ def desc
 
     protected
 
+    def case_sensitive?
```

Comment:
I think the last time I looked into this, there weren't any options that had case sensitive enums 🤔 

Should this just be an invariant to the OptEnum initialize method?

Reply:
Are you asking if we should always enforce that the values are unique when ignoring case or that it's a kwarg? I know we looked into it and saw nothing would need case sensitivity but I wasn't confident that would always be the case in the future.

Reply:
Enforcing that it should be unique when ignoring case, so that this always holds true: "I wasn't confident that would always be the case in the future"

Reply:
It shouldn't need to be enforced because this function will detect if it is necessary. If it is necessary in the future, then it will behave as it does now and the user will have to specify the option exactly as it's defined.

Reply:
Hey @adfoster-r7 just wanted to check in to see if you still had any concerns here?

Reply:
It'd be nice to cache the result of `def case_sensitive?` but not a blocker